src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs

```diff
@@ -491,19 +491,18 @@ private RecommendedSymbols GetSymbolsOffOfConditionalReceiver(ExpressionSyntax o
             // applicable to it as well.
             var unnamedSymbols = _context.IsNameOfContext || excludeInstance
                 ? default
-                : GetUnnamedSymbols(originalExpression);
+                : GetUnnamedSymbols(containerSymbol, originalExpression);
             return new RecommendedSymbols(namedSymbols, unnamedSymbols);
         }
 
-        private ImmutableArray<ISymbol> GetUnnamedSymbols(ExpressionSyntax originalExpression)
+        private ImmutableArray<ISymbol> GetUnnamedSymbols(ISymbol containerSymbol, ExpressionSyntax originalExpression)
```

Comment:
i don't think this is the right approach.  Let me find the approach i think we should take here.

Reply:
@MaStr11  opened https://github.com/dotnet/roslyn/pull/52193 for this.